### PR TITLE
feat(frontend): add balance to TokenUi

### PR DESCRIPTION
**Pull request by AntonioVentilii-DFINITY**
# Motivation

The balance store is called a few times in the code, related to derived store `combinedDerivedEnabledNetworkTokensUi`. So, to simplify the code, we add the prop `balance` to type `TokenUi`.
After this PR, we will make use of the new prop throughout the code.

# Changes

- Add `balance` to `TokenUi`.
- Calculate balance in store `combinedDerivedEnabledNetworkTokensUi`.

# Tests

No changes in local replica, just more loading time (that should be compensated once we use the new prop).

### CODE REVIEW

File: src/frontend/src/lib/types/token.ts [hunk 1]
```diff
@@ -40,6 +40,7 @@ export type OptionTokenStandard = TokenStandard | undefined | null;
 export type TokenToPin = Pick<Token, 'id'> & { network: Pick<Token['network'], 'id'> };
 
 interface TokenFinancialData {
+	balance?: number;
```
Comment (peterpeterparker) [inline src/frontend/src/lib/types/token.ts:43]: I would keep non formatted values in stores, in that case `BigNumber` which feels more future prone. If it's more handy to hold a formatted value in the store then I would name the variable accordingly, like `formattedBalance` or something.

Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/types/token.ts:43]: make sense! however is more useful to have the formatted one: i will rename it, but if you prefer i can add even the `BigNumber` prop, having two new values: `balance` and `formattedBalance`

File: src/frontend/src/lib/types/token.ts [hunk 2]
```diff
@@ -40,6 +40,7 @@ export type OptionTokenStandard = TokenStandard | undefined | null;
 export type TokenToPin = Pick<Token, 'id'> & { network: Pick<Token['network'], 'id'> };
 
 interface TokenFinancialData {
+	formattedBalance?: number;
```
Comment (peterpeterparker) [inline src/frontend/src/lib/types/token.ts:43]: If we want to hold the formatted balance, shouldn't it be a `string` then or is there a use case to convert it to number?

File: src/frontend/src/lib/types/token.ts (hunk 1 above)
Comment (peterpeterparker) [reply src/frontend/src/lib/types/token.ts]: Sounds good

File: src/frontend/src/lib/types/token.ts (hunk 2 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/types/token.ts:43]: Yes, it is even more consistent with the name

**Review by peterpeterparker — APPROVED** [review]
LGTM, thx

**Pull request merged**